### ISSUE TITLE
Revisit storing entire search results in memento

### ISSUE THREAD

**Issue by bpasero** [microsoft/vscode#145997]
Steps to Reproduce:

1. fresh user data dir
2. search for "Copyright" in our project
3. open as search editor
4. close VSCode

=> the workspace state DB is now 1 MB large, looks like it stores the entire search result in it

I suggest to reconsider the storage service for this use and persist to disk via file service. The storage service and mementos were more designed with simple key/value needs in mind.

**Comment by JacksonKearl** [comment]
We keep track of the match ranges so we can restore the highlights, we used to do it always even though we only keep track of the document when it's dirty. I've changed it to only store them when the document is dirty and thus will be persisted, but it's still stored in storage service. Do you have any suggestions for storing this elsewhere?

**Comment by bpasero** [comment]
> Do you have any suggestions for storing this elsewhere?

Yeah on disk. The same is what testing did to get out of storage service:

https://github.com/microsoft/vscode/blob/14a318068c9e7f09f1dab3c20893e22a8a1b18c8/src/vs/workbench/contrib/testing/common/testResultStorage.ts#L229

Just come up with a new folder that does not contain a `.` because extensions also persist there with their extensions ID as folder name.
